src/lib.rs

```diff
@@ -78,7 +78,7 @@ use std::mem::transmute;
 /// <BigEndian as ByteOrder>::write_i16(&mut buf, -50_000);
 /// assert_eq!(-50_000, <BigEndian as ByteOrder>::read_i16(&buf));
 /// ```
-pub trait ByteOrder {
+pub trait ByteOrder : std::marker::PhantomFn<Self> {
```

Comment:
The ideomatic way of writing this is

`pub trait ByteOrder : std::marker::MarkerTrait {`